src/transport_ext.rs

```diff
@@ -31,11 +39,31 @@ pub trait TransportExt: Transport {
     ///
     /// This method returns an `Arc<BandwidthSinks>` that can be used to retreive the total number
     /// of bytes transferred through the sockets.
-    fn with_bandwidth_logging(self) -> (BandwidthLogging<Self>, Arc<BandwidthSinks>)
+    fn with_bandwidth_logging<S>(self) -> (Boxed<(PeerId, StreamMuxerBox)>, Arc<BandwidthSinks>)
     where
-        Self: Sized,
+        Self: Sized + Send + Unpin + 'static,
+        Self::Dial: Send + 'static,
+        Self::ListenerUpgrade: Send + 'static,
+        Self::Error: Send + Sync,
+        Self::Output: Into<(PeerId, S)>,
+        S: StreamMuxer + Send + 'static,
+        S::Substream: Send + 'static,
+        S::Error: Send + Sync + 'static,
     {
-        BandwidthLogging::new(self)
+        let sinks = BandwidthSinks::new();
+        let sinks_copy = sinks.clone();
+        let transport = Transport::map(self, |output, _| {
+            let (peer_id, stream_muxer_box) = output.into();
+            (
+                peer_id,
+                StreamMuxerBox::new(BandwidthLogging::new(
```

Comment:
If we use `StreamMuxerBox` here, then lets make `BandwidthLogging` and `InstrumentStream` `pub(crate)`!